marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/controllers/ModelController.java

```diff
@@ -91,9 +91,13 @@
     }
 
     @RequestMapping(value = "/{modelName}/info", method = RequestMethod.PUT)
+    @ApiImplicitParam(required = true, paramType = "body", dataType = "UpdateModelInfoInput")
     @Secured("ROLE_writeEntityModel")
-    public ResponseEntity<Void> updateModelInfo(@PathVariable String modelName, @RequestBody UpdateModelInfoInput input) {
-        newService().updateModelInfo(modelName, input.description);
+    public ResponseEntity<Void> updateModelInfo(@PathVariable String modelName, @RequestBody @ApiParam(hidden = true) JsonNode input) {
+        if(input.get("name") != null && !(input.get("name").asText().equals(modelName))){
+            throw new RuntimeException("Invalid payload: Incorrect model name: "+ input.get("name").asText());
```

Comment:
We don't want to use the term "payload" here, which is not meaningful to a user. The fact that the model name is incorrect is meaningful, though it appears to be either due to programmer error or malicious use. So a better message would be "Unable to update entity model; incorrect model name: " + name.